File: graphql-subscriptions-rails/app/graphql/mutations/create_post.rb
# frozen_string_literal: true

module Mutations
  class CreatePost < Mutations::MutationSupport
    def resolve(_object, args, _context)
      post = Post.create! do |post|
        post.title = args[:title]
        post.body = args[:body]
      end
      MySchema.subscriptions.trigger('postAdded', {}, post)
      post
    end
  end
end

File: graphql-subscriptions-apollo-express/graphql/resolvers/posts/create_post.js
const {Post} = require('../../../database/models');
const {pubsub} = require('../../../pubsub/local_pubsub');

const createPost = async (_, {title, body}) => {
    const post = await Post.create({title: title, body: body});
    pubsub.publish('postAdded', {postAdded: post});
    return post;
};

module.exports = createPost;
